### ISSUE TITLE
parsed-literal with links and latex

### ISSUE THREAD

**Issue by andreacassioli** [sphinx-doc/sphinx#2026]
Hi,
I am struggling to use parsed-literal with links in Sphinx 1.3.1+. The following simple code 

```
.. parsed-literal::

   public :ref:`fusion.variable`  antidiag()
```

'
It works nicely with the html writer. But with the latex writer it generates the following tex.

```
\begin{alltt}
public {\hyperref[fusion-variable:fusion-variable]{\emph{Variable}}}  antidiag()
\end{alltt}
```

This does not seems to be working with pdflatex:

```
! Missing \endcsname inserted.
<to be read again> 
                   \unhbox 
l.5106 ...yperref[fusion-variable:fusion-variable]
                                                  {\emph{Variable}} antidiag()
?
```

Any clue? It seems to me that removing the link works.

**Comment by andreacassioli** [comment]
Hi,
I still get this error. It may be related with alltt env used by the Tex writer:

http://tex.stackexchange.com/questions/63170/pageref-inside-alltt

Indeed the code I get is 

```
\begin{alltt}
public {\hyperref[fusion-variable:fusion-variable]{\emph{Variable}}}  antidiag()
\end{alltt}
```

**Comment by tk0miya** [comment]
Fixed at 53cd2d4
Thank you for reporting

**Comment by jfbu** [comment]
The fix was only partial.  Consider this

```rst
Welcome to temp's documentation!
================================


.. _fusion.variable:

Section
-------

.. parsed-literal::

   public :ref:`fusion.variable`  antidiag()

```

Then testing latex target with Sphinx 1.5 there is a problem. This problem comes from the comma inside `\DUrole` in produced latex:

```latex
\section{Section}
\label{index:fusion-variable}\label{index:section}\label{index:welcome-to-temp-s-documentation}\begin{alltt}
public {\hyperref[index:fusion\string-variable]{\sphinxcrossref{\DUrole{std,std-ref}{Section}}}}  antidiag()
\end{alltt}
```

The comma is made active by `alltt` latex environment, and this causes error in `\DUrole`.

**Comment by jfbu** [comment]
Fixed (in a new way) at 4b42446 (#3206) and e9c472c (#3208).
